src/NzbDrone.Core/Indexers/NyaaTorrentRssParser.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Linq;
+using System.Xml.Linq;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Indexers.Exceptions;
+
+namespace NzbDrone.Core.Indexers
+{
+    public class NyaaTorrentRssParser : TorrentRssParser
+    {
+        public NyaaTorrentRssParser()
+        {
+        }
+
+        protected override long GetSize(XElement item)
+        {
+            var itemstripped = item.StripNameSpace();
+            if (itemstripped.Element("size") != null)
+            {
+                return ParseSize(itemstripped.Element("size").Value, true);
+            }
+
+            return base.GetSize(item);
+        }
+
+        protected override string GetInfoHash(XElement item)
```

Comment:
Add a config option to `TorrentRssParser`, such as `InfoHashElementName` would allow Sonarr to use it before falling back to parsing it from the Magnet URL.